File: qz/u2/santi/Exam U2/71-100/ExamSpeakers/src/main/java/ec/edu/espe/examspeakers/controller/SpeakerController.java

package ec.edu.espe.examspeakers.controller;

import ec.edu.espe.examspeakers.model.Speaker;
import java.util.List;

/**
 *
 * @author Sjean
 */
public class SpeakerController {
    private SpeakerDAO speakerDAO;

    public SpeakerController(SpeakerDAO speakerDAO) {
        this.speakerDAO = speakerDAO;
    }

    public void createSpeaker(Speaker speaker) {
        speakerDAO.create(speaker);
    }

    public Speaker readSpeaker(String id) {
        return speakerDAO.read(id);
    }

    public void updateSpeaker(Speaker speaker) {
        speakerDAO.update(speaker);
    }

    public void deleteSpeaker(String id) {
        speakerDAO.delete(id);
    }

    public List<Speaker> listAllSpeakers() {
        return speakerDAO.listAll();
    }
}

File: qz/u3/santi/qz1&2/Quizz_Speakers/Quizz_Speakers/controller/speaker_controller.py
from controller.speaker_dao import SpeakerDAO
from model.speaker import Speaker
from typing import List

class SpeakerController:
    def __init__(self, speaker_dao: SpeakerDAO):
        self.speaker_dao = speaker_dao

    def create_speaker(self, speaker: Speaker):
        self.speaker_dao.create(speaker)

    def read_speaker(self, speaker_id: str) -> Speaker:
        return self.speaker_dao.read(speaker_id)

    def update_speaker(self, speaker: Speaker):
        self.speaker_dao.update(speaker)

    def delete_speaker(self, speaker_id: str):
        self.speaker_dao.delete(speaker_id)

    def list_all_speakers(self) -> List[Speaker]:
        return self.speaker_dao.list_all()
